applicationset/services/pull_request/azure_devops.go

```diff
@@ -42,26 +68,98 @@ type AzureDevOpsService struct {
 var (
 	_ PullRequestService       = (*AzureDevOpsService)(nil)
 	_ AzureDevOpsClientFactory = &devopsFactoryImpl{}
+	_ AzureDevOpsClientFactory = &devopsWorkloadIdentityFactoryImpl{}
+	_ AuthProvider             = &PatAuthProvider{}
+	_ AuthProvider             = &WorkloadIdentityAuthProvider{}
+	_ AuthProvider             = &AnonymousAuthProvider{}
 )
 
-func NewAzureDevOpsService(token, url, organization, project, repo string, labels []string) (PullRequestService, error) {
-	organizationURL := buildURL(url, organization)
+// AuthProvider defines the interface for Azure DevOps authentication methods
+type AuthProvider interface {
+	CreateClientFactory(ctx context.Context, organizationURL string) (AzureDevOpsClientFactory, error)
```

Comment:
Why do we need the context here?
I see that it's not used by the three interface implementations. 

Am I missing anything?

Reply:
Thanks, updated